test/ibmq/test_ibmq_job.py

```diff
@@ -213,10 +213,10 @@ def test_retrieve_job_uses_appropriate_backend(self, backend: Backend):
         if not backend_2:
             raise SkipTest('Skipping test that requires multiple backends')
 
-        job_1 = self._sim_job(backend=backend_1, qc=self.bell)
-        job_1.wait_for_final_state()
-        job_2 = self._sim_job(backend=backend_2, qc=self.bell)
-        job_2.wait_for_final_state()
+        job_1 = self._run_job(backend=backend_1, qc=self.bell)
+        job_1.cancel()
```

Comment:
Similar to my comment above. `job.cancel()` may fail if it's not in a cancellable state - use the `cancel_job` method instead (which is basically best effort).

Reply:
**Update:** There are now no references of `job.cancel()` in the PR.